google-login-plugin/src/com/google/gct/login/stats/UsageTrackerService.java

```diff
@@ -15,52 +15,28 @@
  */
 package com.google.gct.login.stats;
 
-import com.intellij.openapi.util.KeyedExtensionCollector;
-import com.intellij.util.PlatformUtils;
+import com.intellij.openapi.components.ServiceManager;
 
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
- * Usage Tracker Service for obtaining extensions declared in plugin.xml with {@link
- * UsageTrackerExtensionPointBean}
+ * Usage Tracker Service for obtaining UsageTracking implementations
  */
-public final class UsageTrackerService {
+public abstract class UsageTrackerService {
 
-  private static final KeyedExtensionCollector<UsageTracker, String> COLLECTOR =
-      new KeyedExtensionCollector<UsageTracker, String>(
-          UsageTrackerExtensionPointBean.EP_NAME.getName());
-  private static final UsageTrackerManager usageTrackerManager = UsageTrackerManager.getInstance();
-
-  /**
-   * When using the usage tracker, do NOT include any information that can identify the user
-   *
-   * @return the usage tracker for the current platform
-   */
   @NotNull
   public static UsageTracker getInstance() {
-    String key = PlatformUtils.getPlatformPrefix();
-    return getInstance(key);
+    return ServiceManager.getService(UsageTrackerService.class).getTracker();
   }
 
   /**
-   * For usage tracking in the Cloud Tools family of plugins, we only want one tracker pinging an
-   * analytics backend for a given platform. New platform specific trackers can register with their
-   * platform prefix key and the UsageTrackerService will select them.  Otherwise, the no-op usage
-   * tracker will be used.
-   *
-   * @param key A string search key associated with an extension
-   * @return the first implementation of UsageTracker associated with {@param key}
+   * Do not return a tracker that includes PII
    */
-  static UsageTracker getInstance(@Nullable String key) {
-    UsageTracker instance = (key == null) ? null : COLLECTOR.findSingle(key);
-    if (instance != null) {
-      return instance;
-    }
-    return new NoOpUsageTracker();
-  }
+  @NotNull
+  protected abstract UsageTrackerServiceImpl.UsageTracker getTracker();
 
-  // Interface for UsageTracker implementations defined in other plugins
+  // Interface for UsageTracker implementation
```

Comment:
this comment is a little redundant.  It should describe what objects of this type do.